enginetest/queries/queries.go

```diff
@@ -9354,6 +9354,84 @@ from typestable`,
 			{false},
 		},
 	},
+	{
+		Query: `SELECT json_type('{"a": [10, true, "abc"]}');`,
+		Expected: []sql.Row{
+			{"OBJECT"},
+		},
+	},
+	{
+		Query: `SELECT json_type(json_extract('{"a": [10, true, "abc"]}', '$.a'));`,
+		Expected: []sql.Row{
+			{"ARRAY"},
+		},
+	},
+	{
+		Query: `SELECT json_type(json_extract('{"a": [10, true, "abc"]}', '$.a[0]'));`,
+		Expected: []sql.Row{
+			{"INTEGER"},
+		},
+	},
+	{
+		Query: `SELECT json_type(json_extract('{"a": [10, true, "abc"]}', '$.a[1]'));`,
+		Expected: []sql.Row{
+			{"BOOLEAN"},
+		},
+	},
+	{
+		Query: `SELECT json_type(json_extract('{"a": [10, true, "abc"]}', '$.a[2]'));`,
+		Expected: []sql.Row{
+			{"STRING"},
+		},
+	},
+	{
+		Query: `SELECT json_type(json_extract('{"a": 123.456}', '$.a'));`,
+		Expected: []sql.Row{
+			{"DOUBLE"},
+		},
+	},
+	{
+		Query: `SELECT json_type(json_extract('{"a": null}', '$.a'));`,
+		Expected: []sql.Row{
+			{"NULL"},
+		},
+	},
+	{
+		Query: "SELECT json_type(cast(cast('2001-01-01 12:34:56.123456' as datetime) as json));",
+		Expected: []sql.Row{
+			{"DATETIME"},
+		},
+	},
+	{
+		Query: "SELECT json_type(cast(cast('2001-01-01 12:34:56.123456' as date) as json));",
+		Expected: []sql.Row{
+			{"DATE"},
+		},
+	},
+	{
+		Query: "select json_type(cast(cast(1 as unsigned) as json));",
+		Expected: []sql.Row{
+			{"UNSIGNED INTEGER"},
```

Comment:
oooh, nice work! The docs don't mention "UNSIGNED INTEGER" as a possible response, but that's exactly what MySQL returns! Good job reversing engineering MySQL's behavior here.